stats/view/aggregation_data.go

```diff
@@ -139,26 +159,61 @@ func (a *DistributionData) addSample(f float64) {
 	a.SumOfSquaredDev = a.SumOfSquaredDev + (f-oldMean)*(f-a.Mean)
 }
 
-func (a *DistributionData) incrementBucketCount(f float64) {
-	if len(a.bounds) == 0 {
-		a.CountPerBucket[0]++
+func (a *DistributionData) addToBucket(val float64, ex *exemplar.Exemplar) {
+	var (
+		count  *int64
+		bucket *Bucket
+	)
+
+	for i, b := range a.bounds {
+		if val < b {
+			count = &a.CountPerBucket[i]
```

Comment:
I now wonder why bounds, Bucket aren't of type `map[int]float64`.
I believe it would save on:
* Potential runtime crashes from out-of-bounds index access with slices
* Having to understand that the zeroth bucket is for the case when we don't have any bounds
* Save on the pointer manipulation
* Save on upfront memory allocations e.g. if we had a program using lots of views with DistributionData
* [index] access would automatically create zero values

To illustrate this point example this function would simply become
```go
func (a *DistributionData) addToBucket(val float64, ex *exemplar.Exemplar) {
        valBucketIndex := len(a.bounds)
        for i, b := range a.bounds {
                if val < b {
                        valBucketIndex = i 
                        break
                }
        }

        bucket := a.Buckets[valBucketIndex]
        bucket.Count += 1
        a.CountPerBucket[valBucketIndex] = bucket.Count
        bucket.maybeRetainExemplar(val, ex)
}
```

Although perhaps it is too late to change the types of Count* and Bucket* but it is something to keep in mind for future designs :)